src/test/java/com/ericsson/ei/subscription/SubscriptionValidatorTest.java

```diff
@@ -71,64 +72,52 @@ public SubscriptionValidatorTest() {
         NotificationMessageKeyValue notificationMessageKeyValueinvalid = new NotificationMessageKeyValue();
         notificationMessageKeyValueinvalid.setFormkey("");
         notificationMessageKeyValueinvalid.setFormvalue("@");
-        subscriptionInvalid.getNotificationMessageKeyValues().add(notificationMessageKeyValueinvalid);
+        subscriptionInvalid.getNotificationMessageKeyValues()
+                           .add(notificationMessageKeyValueinvalid);
 
         subscriptionInvalid.setNotificationMeta("kalle1.kal  le2@domain.com");
         subscriptionInvalid.setNotificationType("MAIL");
         subscriptionInvalid.setRepeat(true);
         Requirement requirementInvalid = new Requirement();
         Condition conditionInvalid = new Condition();
-        conditionInvalid.setJmespath("split(data.identity', '/') | [2] =='com.mycompany.myproduct'");
+        conditionInvalid.setJmespath(
+                "split(data.identity', '/') | [2] =='com.mycompany.myproduct'");
         requirementInvalid.getConditions().add(conditionInvalid);
         subscriptionInvalid.getRequirements().add(requirementInvalid);
     }
 
     /**
-     * Validator unit tests for SubscriptionName parameter in Subscription. Valid
-     * "SubscriptionName" values: - All letters and numbers: [A-Za-z0-9_].
+     * Validator unit tests for SubscriptionName parameter in Subscription. Valid "SubscriptionName"
+     * values: - All letters and numbers: [A-Za-z0-9_].
      */
     @Test
     public void validateSubscriptionNameValidNameTest() throws Exception {
-        String subscriptionName = "Kalle1";
-        try {
-            invokeMethod(SubscriptionValidator.class, "validateSubscriptionName", subscriptionName);
-        } catch (SubscriptionValidationException e) {
-            assertTrue(e.getMessage(), false);
-            return;
-        }
-    }
-
-    @Test
-    public void validateSubscriptionNameValidNameTest2() throws Exception {
-        String subscriptionName = "Kalle_1";
-        try {
-            invokeMethod(SubscriptionValidator.class, "validateSubscriptionName", subscriptionName);
-        } catch (SubscriptionValidationException e) {
-            assertTrue(e.getMessage(), false);
-            return;
-        }
-    }
-
-    @Test
-    public void validateSubscriptionNameInvalidNameTest() throws Exception {
-        String subscriptionName = "Kal--l[[e1";
-        try {
-            invokeMethod(SubscriptionValidator.class, "validateSubscriptionName", subscriptionName);
-        } catch (SubscriptionValidationException e) {
-            return;
+        String[] validSubscriptionNameList = { "Kalle", "kalle", "KALLE", "kall_e", "Kalle1",
```

Comment:
Here and elsewhere, is there a reason not to use `@Parameterized` ([example](https://www.tutorialspoint.com/junit/junit_parameterized_test.htm)) and not catch the exception

Reply:
It is complicated and need to be run with another runner than we use for this test.

Reply:
I was only wondering as you have added the for loop. My view is not to add your own logic if there is something available.